### PR TITLE
Remove qb-menu and qb-target

### PR DESCRIPTION
**Pull request by JoeSzymkowiczFiveM**
**Describe Pull request**
Shift taxijob to ox resources, remove Polyzone and qb-target, qb-menu. Retaining existing functionality prior to push.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? yes
- Does your code fit the style guidelines? yes
- Does your PR fit the contribution guidelines? yes

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -35,12 +35,50 @@ local NpcData = {
...
 
+local CabParkingZone = nil
+local CabBossZone = nil
+
+local taxiPed = nil
+
 -- events
+
+AddEventHandler('onResourceStop', function(resourceName)
+    if resourceName == GetCurrentResourceName() and QBCore.Functions.GetPlayerData() ~= {} then
```
Comment (BerkieBb) [inline client/main.lua:46]: `~= {}` doesn't work, instead use `table.type(QBCore.Functions.GetPlayerData()) ~= 'empty'`

Comment (BerkieBb) [reply client/main.lua:46]: And you can invert both if statements here and combine them into `if resource ~= GetCurrentResourceName() or table.type(QBCore.Functions.GetPlayerData()) == 'empty' or not Config.UseTarget then return end`

File: client/main.lua [hunk 2]
```diff
@@ -35,12 +35,50 @@ local NpcData = {
...
+        if Config.UseTarget then
+            DeletePed(taxiPed)
+        end
+        
+    end
+end)
+AddEventHandler('onResourceStart', function(resourceName)
```
Comment (BerkieBb) [inline client/main.lua:53]: Same as above

File: client/main.lua [hunk 3]
```diff
@@ -345,60 +366,58 @@ RegisterNetEvent('qb-taxi:client:DoTaxiNpc', function()
...
                         end
-                    end
-
-                    Wait(1)
-                end
-            end)
 
-
-           end
+                        Wait(1)
```
Comment (BerkieBb) [inline client/main.lua:417]: Reduce to wait 0

File: client/main.lua [hunk 4]
```diff
@@ -345,60 +366,58 @@ RegisterNetEvent('qb-taxi:client:DoTaxiNpc', function()
...
-                                })
-                                ClearPedTasksImmediately(NpcData.Npc)
-                                FreezeEntityPosition(NpcData.Npc, false)
-                                TaskEnterVehicle(NpcData.Npc, veh, -1, freeSeat, 1.0, 0)
-                                QBCore.Functions.Notify(Lang:t("info.go_to_location"))
-                                if NpcData.NpcBlip ~= nil then
-                                    RemoveBlip(NpcData.NpcBlip)
+                                    meterIsOpen = true
+                                    meterActive = true
+                                    lastLocation = GetEntityCoords(PlayerPedId())
```
Comment (BerkieBb) [inline client/main.lua:395]: Use ped here

File: client/main.lua [hunk 5]
```diff
@@ -236,45 +273,34 @@ local function calculateFareAmount()
...
 
--- qb-menu
-
 function TaxiGarage()
-    local vehicleMenu = {
-        {
-            header = Lang:t("menu.taxi_menu_header"),
-            isMenuHeader = true
-        }
+    local resgisteredMenu = {
```
Comment (BerkieBb) [inline client/main.lua:277]: resgistered

File: client/main.lua [hunk 6]
```diff
@@ -236,45 +273,34 @@ local function calculateFareAmount()
...
 
-    vehicleMenu[#vehicleMenu+1] = {
-        header = Lang:t("menu.close_menu"),
-        txt = "",
-        params = {
-            event = "qb-menu:client:closeMenu"
-        }
-    }
-    exports['qb-menu']:openMenu(vehicleMenu)
+    resgisteredMenu["options"] = options
```
Comment (BerkieBb) [inline client/main.lua:301]: Just .options is fine

File: client/main.lua [hunk 7]
```diff
@@ -345,60 +366,58 @@ RegisterNetEvent('qb-taxi:client:DoTaxiNpc', function()
...
-
-                                for i=maxSeats - 1, 0, -1 do
-                                    if IsVehicleSeatFree(veh, i) then
-                                        freeSeat = i
-                                        break
+            if not Config.UseTarget then
+                CreateThread(function()
+                    while not NpcData.NpcTaken do
+
+                        local ped = PlayerPedId()
```
Comment (BerkieBb) [inline client/main.lua:373]: You can use cache.ped instead

File: client/main.lua [hunk 8]
```diff
@@ -345,60 +366,58 @@ RegisterNetEvent('qb-taxi:client:DoTaxiNpc', function()
...
+                        local pos = GetEntityCoords(ped)
+                        local dist = #(pos - vector3(Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].x, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].y, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].z))
+
+                        if dist < 20 then
+                            DrawMarker(2, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].x, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].y, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.3, 0.3, 255, 255, 255, 255, 0, 0, 0, 1, 0, 0, 0)
+
+                            if dist < 5 then
+                                DrawText3D(Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].x, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].y, Config.NPCLocations.TakeLocations[NpcData.CurrentNpc].z, Lang:t("info.call_npc"))
+                                if IsControlJustPressed(0, 38) then
+                                    local veh = GetVehiclePedIsIn(ped, 0)
```
Comment (BerkieBb) [inline client/main.lua:383]: You can use cache.vehicle instead

File: client/main.lua [hunk 9]
```diff
@@ -509,111 +528,78 @@ RegisterNetEvent('qb-taxijob:client:requestcab', function()
...
-                        label = '🚕 Request Taxi Cab',
-                        job = "taxi",
-                    }
-                },
-            distance = 2.5,
-            },
-            spawnNow = true,
-            currentpednumber = 0,
-          })
+        RequestModel(`a_m_m_indian_01`)
```
Comment (BerkieBb) [inline client/main.lua:535]: Instead of these 4 lines, use lib.requestModel

File: client/main.lua [hunk 10]
```diff
@@ -509,111 +528,78 @@ RegisterNetEvent('qb-taxijob:client:requestcab', function()
...
     end)
 end
 
 local zone
 local delieveryZone
 
+function onEnterCallZone()
+    if whitelistedVehicle() and not isInsidePickupZone and not NpcData.NpcTaken then
+        isInsidePickupZone = true
+        exports['qb-core']:DrawText(Lang:t("info.call_npc"), Config.DefaultTextLocation)
```
Comment (BerkieBb) [inline client/main.lua:563]: Use lib.drawText instead

File: client/main.lua [hunk 11]
```diff
@@ -707,76 +693,105 @@ function dropNpcPoly()
                     end
                     RemovePed(NpcData.Npc)
                     ResetNpcTask()
-                    delieveryZone:destroy()
+                    delieveryZone:remove()
                     break
                 end
             end
             Wait(1)
```
Comment (BerkieBb) [inline client/main.lua:700]: wait 0

**Review by BerkieBb — CHANGES_REQUESTED** [review]

**Comment by BerkieBb** [comment]
Aside from my reviews, make local functions out of the current global ones as I don't think they're used globally, invert if statements to reduce indentation and fix the issues with the linter

File: client/main.lua (hunk 3 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Updated this.

File: client/main.lua (hunk 4 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Updated to use cache.

File: client/main.lua (hunk 5 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Corrected misspelling

File: client/main.lua (hunk 7 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Updated to use cache

File: client/main.lua (hunk 8 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Updated to use cache.

File: client/main.lua (hunk 11 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Updated interval

File: client/main.lua (hunk 10 above)
Comment (JoeSzymkowiczFiveM) [reply client/main.lua]: Updated to use lib.showTextUI

**Comment by ChatDisabled** [comment]
@YvanCywan I'll overrule that for now 😃

**Pull request merged**